coreppl/dist.mc

```diff
@@ -52,6 +52,32 @@ lang BetaDist = Dist
     else never
 end
 
+-- DCategorical {p=p} is equivalent to DMultinomial {n=1, p=p}
+lang CategoricalDist = Dist
+
+  syn Dist =
+  | DCategorical { p: Expr }
```

Comment:
I'm not sure how explicit we should be with the arguments for distributions. For example `DBeta` now has two (explicit) arguments: `a` and `b`. We could equally well have a single expression `p` as argument for `DBeta` (similar to `DCategorical`), which we restrict to have type `{a: Float, b: Float}` in the type system.

Maybe the idea is to have explicit arguments only when there is a fixed number of arguments? For consistency, maybe we should just have all distributions take a single argument, and then handle everything in the type system.